In [151]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import mode
import statsmodels.api as sm

In [152]:
df = pd.read_csv("datos_whales.csv")

In [153]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 701 entries, 0 to 700
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Country      701 non-null    object 
 1   SeeWhales    701 non-null    int64  
 2   Age          691 non-null    float64
 3   Gender       701 non-null    object 
 4   Education    701 non-null    object 
 5   Income_AUD   674 non-null    object 
 6   Income       674 non-null    float64
 7   AdultWhales  701 non-null    int64  
 8   YoungWhales  701 non-null    object 
 9   Max_wtp      701 non-null    float64
 10  wtp          701 non-null    object 
 11  PayConserve  701 non-null    int64  
dtypes: float64(3), int64(3), object(6)
memory usage: 65.8+ KB


###Disposición a pagar por conservar el abistamiento de ballenas

In [154]:
df = df[["Education","Income", "Age", "SeeWhales","PayConserve", "Income_AUD"]]

In [155]:
df

Education   Income   Age  SeeWhales  PayConserve  \
0                 Diploma  65000.5  31.0          1            1   
1                  Degree  10000.0  22.0          0            0   
2                  Degree  85000.5  46.0          0            0   
3                 Diploma  25000.5  37.0          0            1   
4                  Degree      NaN  66.0          0            0   
..                    ...      ...   ...        ...          ...   
696     Completed year 10  10000.0  56.0          0            1   
697     Completed year 10  35000.5  38.0          0            1   
698               Diploma  25000.5  24.0          0            1   
699  Post-graduate degree  35000.5  50.0          0            1   
700                Degree  85000.5  32.0          0            0   

            Income_AUD  
0    $60,001 - $70,000  
1    Less than $20,000  
2    More than $70,000  
3    $20,001 - $30,000  
4                  NaN  
..                 ...  
696  Less than $20,000  
697  $30,001 - $40,000  
698  $20,001 - $30,000  
699  $30,001 - $40,000  
700  More than $70,000  

[701 rows x 6 columns]

In [156]:
df.describe()

Income         Age   SeeWhales  PayConserve
count    674.000000  691.000000  701.000000   701.000000
mean   44540.456973   42.918958    0.126961     0.660485
std    26362.209685   15.562472    0.333168     0.473883
min    10000.000000   13.000000    0.000000     0.000000
25%    25000.500000   30.000000    0.000000     0.000000
50%    35000.500000   41.000000    0.000000     1.000000
75%    65000.500000   54.500000    0.000000     1.000000
max    85000.500000   82.000000    1.000000     1.000000

In [157]:
df.mean()

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  """Entry point for launching an IPython kernel.


Income         44540.456973
Age               42.918958
SeeWhales          0.126961
PayConserve        0.660485
dtype: float64

In [158]:
df["Age"] = df["Age"].fillna(42.918958) 

In [159]:
df

Education   Income   Age  SeeWhales  PayConserve  \
0                 Diploma  65000.5  31.0          1            1   
1                  Degree  10000.0  22.0          0            0   
2                  Degree  85000.5  46.0          0            0   
3                 Diploma  25000.5  37.0          0            1   
4                  Degree      NaN  66.0          0            0   
..                    ...      ...   ...        ...          ...   
696     Completed year 10  10000.0  56.0          0            1   
697     Completed year 10  35000.5  38.0          0            1   
698               Diploma  25000.5  24.0          0            1   
699  Post-graduate degree  35000.5  50.0          0            1   
700                Degree  85000.5  32.0          0            0   

            Income_AUD  
0    $60,001 - $70,000  
1    Less than $20,000  
2    More than $70,000  
3    $20,001 - $30,000  
4                  NaN  
..                 ...  
696  Less than $20,000  
697  $30,001 - $40,000  
698  $20,001 - $30,000  
699  $30,001 - $40,000  
700  More than $70,000  

[701 rows x 6 columns]

In [160]:
pd.crosstab(df["Education"],df["PayConserve"])

PayConserve                0    1
Education                        
Completed year 10         42   58
Completed year 12         28   72
Degree                    57  115
Diploma                   28   60
Other - Specify            0    1
Post-graduate degree      24   61
Primary only              14   11
Some secondary schooling  22   33
Trade certificate         23   52

In [161]:
from scipy.stats import chi2_contingency

In [162]:
chi2_contingency(x)

(13.127920849377242,
 0.10752339615615682,
 8,
 array([[ 33.95149786,  66.04850214],
        [ 33.95149786,  66.04850214],
        [ 58.39657632, 113.60342368],
        [ 29.87731812,  58.12268188],
        [  0.33951498,   0.66048502],
        [ 28.85877318,  56.14122682],
        [  8.48787447,  16.51212553],
        [ 18.67332382,  36.32667618],
        [ 25.4636234 ,  49.5363766 ]]))

In [163]:
##El P valor es mayor a 0.05, por ende, no hay relación entre las dos variables

In [164]:
y= pd.crosstab(df["Income_AUD"],df["PayConserve"])

In [165]:
chi2_contingency(y)

(25.6142313740965,
 0.00026265667062334315,
 6,
 array([[28.91394659, 58.08605341],
        [38.55192878, 77.44807122],
        [20.60534125, 41.39465875],
        [26.58753709, 53.41246291],
        [16.9495549 , 34.0504451 ],
        [45.86350148, 92.13649852],
        [46.52818991, 93.47181009]]))

In [166]:
##El P valor es menor a 0.05, por ende, si hay relación entre las dos variables

In [167]:
z= pd.crosstab(df["SeeWhales"],df["PayConserve"])

In [168]:
chi2_contingency(z)

(5.418812987362997, 0.019920898752360208, 1, array([[207.7831669, 404.2168331],
        [ 30.2168331,  58.7831669]]))

In [169]:
##El P valor es menor a 0.05, por ende, si hay relación entre las dos variables

In [170]:
df

Education   Income   Age  SeeWhales  PayConserve  \
0                 Diploma  65000.5  31.0          1            1   
1                  Degree  10000.0  22.0          0            0   
2                  Degree  85000.5  46.0          0            0   
3                 Diploma  25000.5  37.0          0            1   
4                  Degree      NaN  66.0          0            0   
..                    ...      ...   ...        ...          ...   
696     Completed year 10  10000.0  56.0          0            1   
697     Completed year 10  35000.5  38.0          0            1   
698               Diploma  25000.5  24.0          0            1   
699  Post-graduate degree  35000.5  50.0          0            1   
700                Degree  85000.5  32.0          0            0   

            Income_AUD  
0    $60,001 - $70,000  
1    Less than $20,000  
2    More than $70,000  
3    $20,001 - $30,000  
4                  NaN  
..                 ...  
696  Less than $20,000  
697  $30,001 - $40,000  
698  $20,001 - $30,000  
699  $30,001 - $40,000  
700  More than $70,000  

[701 rows x 6 columns]

In [188]:
df.drop(columns=["Income_AUD","Education"])

Income   Age  SeeWhales  PayConserve
0    65000.500000  31.0          1            1
1    10000.000000  22.0          0            0
2    85000.500000  46.0          0            0
3    25000.500000  37.0          0            1
4    44540.456973  66.0          0            0
..            ...   ...        ...          ...
696  10000.000000  56.0          0            1
697  35000.500000  38.0          0            1
698  25000.500000  24.0          0            1
699  35000.500000  50.0          0            1
700  85000.500000  32.0          0            0

[701 rows x 4 columns]

In [189]:
df["Income"] = df["Income"].fillna(44540.456973)

In [190]:
af=df.drop(columns=["Income_AUD","Education"])

In [191]:
X = af[["Income", "Age", "SeeWhales"]]
y = af["PayConserve"]
X = sm.add_constant(X)
model = sm.OLS(y,X)
reg = model.fit()
reg.summary()

/usr/local/lib/python3.7/dist-packages/statsmodels/tsa/tsatools.py:117: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:            PayConserve   R-squared:                       0.021
Model:                            OLS   Adj. R-squared:                  0.016
Method:                 Least Squares   F-statistic:                     4.870
Date:                Thu, 19 May 2022   Prob (F-statistic):            0.00233
Time:                        22:34:27   Log-Likelihood:                -463.40
No. Observations:                 701   AIC:                             934.8
Df Residuals:                     697   BIC:                             953.0
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.8071      0.063     12.740      0.000       0.683       0.932
Income     -4.796e-07   6.91e-07     -0.695      0.488   -1.84e-06    8.76e-07
Age           -0.0034      0.001     -2.890      0.004      -0.006      -0.001
SeeWhales      0.1507      0.054      2.802      0.005       0.045       0.256
==============================================================================
Omnibus:                     5960.352   Durbin-Watson:                   1.897
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              114.305
Skew:                          -0.656   Prob(JB):                     1.51e-25
Kurtosis:                       1.519   Cond. No.                     1.84e+05
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.84e+05. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

Conclusión: ver ballenas incrementa en un 15% la probabilidad de la disposición a pagar dos dólares para preservar el abistamiento de ballenas. Por otro lado, el ingreso disminuye la probabilidad de la disposición a pagar. Y la edad tambien la disminuye en un 0,3%. Todas las variables son significativas excepto el ingreso. 